spec.html

```diff
@@ -59,12 +59,14 @@ <h1>
       </dl>
       <emu-alg>
         1. <ins>Let _slots_ be &laquo; [[ArrayBufferData]], [[ArrayBufferByteLength]], [[ArrayBufferDetachKey]] &raquo;.</ins>
-        1. <ins>If _maxByteLength_ is present, append [[ArrayBufferMaxByteLength]] to _slots_.</ins>
+        1. <ins>If _maxByteLength_ is present and not ~empty~, then</ins>
+          1. <ins>If _byteLength_ &gt; _maxByteLength_, throw a *RangeError* exception.</ins>
+          1. <ins>Append [[ArrayBufferMaxByteLength]] to _slots_.</ins>
         1. Let _obj_ be ? OrdinaryCreateFromConstructor(_constructor_, *"%ArrayBuffer.prototype%"*, <del>&laquo; [[ArrayBufferData]], [[ArrayBufferByteLength]], [[ArrayBufferDetachKey]] &raquo;</del><ins>_slots_</ins>).
         1. Let _block_ be ? CreateByteDataBlock(_byteLength_).
         1. Set _obj_.[[ArrayBufferData]] to _block_.
         1. Set _obj_.[[ArrayBufferByteLength]] to _byteLength_.
-        1. <ins>If _maxByteLength_ is present, then</ins>
+        1. <ins>If _maxByteLength_ is present and not ~empty~, then</ins>
           1. <ins>Assert: _byteLength_ &le; _maxByteLength_.</ins>
```

Comment:
I didn't remove this assert but I'm happy to do so if it's not needed with the check above.

Reply:
Yep, I'll remove it after merging your PR.